5.3.js

```diff
@@ -0,0 +1,37 @@
+function average(array) {
+    function plus(a, b) {
+        return a + b;
+    }
+    return array.reduce(plus) / array.length;
+}
+
+function groupBy(array, groupOf) {
+    var groups = {};
+    array.forEach(function(element) {
+        var groupName = groupOf(element);
+
+        if (!Array.isArray(groups[groupName])) {
+            groups[groupName] = [];
+        }
+
+        if (groupName in groups) {

```

Comment:
po co Ci tak wlasciwe to sprawdzanie czy groupName wystepuje w groups? i uzaleznianie od tego 2ch roznych flow skoro w linii 13 upewnilas sie, ze musi istniec (w dodatku jako array) wiec tutaj moglabys zostawic tylko kod z linii 18 i byloby wystarczajaco